### PR TITLE
Add support for Python 3.10

### PR DESCRIPTION
**Pull request by hofaflo**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,12 +1,12 @@
 [build-system]
-requires = ["setuptools", "wheel", "numpy<1.21.5"]
+requires = ['setuptools', 'wheel', 'oldest-supported-numpy']
```
[the remark this answers is not in the document]
Comment (hofaflo) [reply pyproject.toml:2]: The packages specified here are not used for testing but for building. If we used the newest NumPy for building the C extension, importing it in an environment with an older one won't work if there are incompatible changes in the NumPy C-API (see #58). Working on holgern/pyedflib#164 as a fix for the related issue holgern/pyedflib#156, i found  [`oldest-supported-numpy`](https://pypi.org/project/oldest-supported-numpy/), which should be a cleaner solution than maintaining an upper limit for numpy here.

[the remark this answers is not in the document]
Comment (cbrnr) [reply pyproject.toml:2]: And how is this oldest version determined?

[the remark this answers is not in the document]
Comment (hofaflo) [reply pyproject.toml:2]: Specified [here](https://github.com/scipy/oldest-supported-numpy/blob/main/setup.cfg) based on the platform and Python version.

[the remark this answers is not in the document]
Comment (cbrnr) [reply pyproject.toml:2]: Just to make sure, if we build wheels with the oldest supported NumPy, can I still use it with the latest NumPy (i.e. 1.22)?

[the remark this answers is not in the document]
Comment (hofaflo) [reply pyproject.toml:2]: Yes, just verified this locally for Python 3.8 and 3.10 :)

**Comment by cbrnr** [comment]
Thanks @hofaflo!

**Pull request merged**